### ISSUE TITLE
docker push not working in 1.9.1

### ISSUE THREAD

**Issue by zopyx** [docker/docker#18526]
Why is this happening???

```
[docker@vmd9666 docker-plone5]$ docker version
Client:
 Version:         1.9.1-fc23
 API version:     1.21
 Package version: docker-1.9.1-2.git78bc3ea.fc23.x86_64
 Go version:      go1.5.1
 Git commit:      f7c1d52-dirty
 Built:           Fri Nov 20 21:07:14 UTC 2015
 OS/Arch:         linux/amd64

Server:
 Version:         1.9.1-fc23
 API version:     1.21
 Package version: docker-1.9.1-2.git78bc3ea.fc23.x86_64
 Go version:      go1.5.1
 Git commit:      f7c1d52-dirty
 Built:           Fri Nov 20 21:07:14 UTC 2015
 OS/Arch:         linux/amd64
[docker@vmd9666 docker-plone5]$ docker push zopyx/plone-50-demo
The push refers to a repository [zopyx/plone-50-demo] (len: 1)
unable to ping registry endpoint https://zopyx/v0/
v2 ping attempt failed with error: Get https://zopyx/v2/: dial tcp: lookup zopyx: No address associated with hostname
 v1 ping attempt failed with error: Get https://zopyx/v1/_ping: dial tcp: lookup zopyx: No address associated with hostname
```

**Comment by thaJeztah** [comment]
Looks like you're running a RedHat build of Docker. (The git commit of that build is not in this repository, but can be [found here](https://github.com/rhatdan/docker/commit/78bc3eaee7a4639325ca103154a69f91ddb1a093), and the changes in that commit are in a file [`credentials_linux.go`](https://github.com/rhatdan/docker/blob/78bc3eaee7a4639325ca103154a69f91ddb1a093/api/server/credentials_linux.go) that [doesn't exist here](https://github.com/docker/docker/tree/v1.9.1/api/server))

Perhaps this is related to https://github.com/docker/docker/issues/18406, can you open this issue in the RedHat issue tracker in stead? https://bugzilla.redhat.com/ Thanks!

**Comment by calavera** [comment]
Hi @zopyx!

I'm very sorry you're having issues. Unfortunately, we cannot assist you when a vendor modify and ships a different version of docker, because we don't really know their reasoning for those changes.

I'm sure you'll get more help if you contact RedHat directly. I'm going to close this issue so other maintainers don't get dragged into it.

Cheers,
David

**Comment by macropin** [comment]
I've just experienced push failures with docker-1.9.1-4.git6ec29ef.fc23.x86_64:

> [root@localhost ~]# docker push macropin/nginx-ingress-thirdparty
> The push refers to a repository [docker.io/macropin/nginx-ingress-thirdparty] (len: 0)
> 2b2efeef7242: Preparing 
> unauthorized: access to the requested resource is not authorized

The problem is docker login does not store the correct address in ~/.docker/config.json. If you manually replace "docker.io" with "https://index.docker.io/v1/" the push will succeed.

I'm guessing this is a regression due to some out of band patches that Red Hat are maintaining.

I bumped the original Fedora ticket https://bugzilla.redhat.com/show_bug.cgi?id=1289963
